src/shared/Microsoft.Git.CredentialManager/Authentication/MicrosoftAuthentication.cs

```diff
@@ -197,6 +197,14 @@ private async Task<IPublicClientApplication> CreatePublicClientApplicationAsync(
                 appBuilder.WithLogging(OnMsalLogMessage, LogLevel.Verbose, enablePiiLogging, false);
             }
 
+            // If we have a parent window ID we should tell MSAL about it so it can parent any authentication dialogs
+            // correctly. We only support this on Windows right now as MSAL only supports embedded/dialogs on Windows.
+            if (PlatformUtils.IsWindows() && !string.IsNullOrWhiteSpace(Context.Settings.ParentWindowId) &&
+                int.TryParse(Context.Settings.ParentWindowId, out int hWndInt) && hWndInt > 0)
```

Comment:
Nice. This handles `INVALID_HANDLE_VALUE` as well: It is `-1`, cast to `HANDLE`: https://github.com/git-for-windows/git-sdk-64/blob/8be5d7823baeb3ff55a4aba9254fd3ca4e96e662/mingw64/x86_64-w64-mingw32/include/handleapi.h#L16

Reply:
MSAL only wants to be told about "real" handles. Some library versions don't handle bad/invalid handles well.